### PR TITLE
* Add additional use cases to SecretsManagerSecretEngine

### PR DESCRIPTION
**Pull request by mayitbeegh**
* Add unit tests

### CODE REVIEW

**Comment by mayitbeegh** [comment]
Context:
SecretsManager allows 2 types of encrypted data: string and binary. String usually takes the form of key-value pairs.

Before this PR, the engine is able to:
- return decrypted value when the encrypted data is a key-value string and a key is provided
- return decrypted value when the encrypted data is a binary and a key is NOT provided

With this PR, the engine is able to:
- return decrypted value regardless of the data type of the encrypted data if config starts with `encryptedFile`
- return decrypted value when the encrypted data is a key-value string and a key is provided
- return decrypted value when the encrypted data is a plaintext string and a key is NOT provided

File: kork-secrets-aws/kork-secrets-aws.gradle [hunk 1]
```diff
@@ -8,4 +8,7 @@ dependencies {
 
   implementation "com.amazonaws:aws-java-sdk-s3"
   implementation "com.amazonaws:aws-java-sdk-secretsmanager"
+
+  testImplementation "org.spockframework:spock-core"
```
Comment (fieldju) [inline kork-secrets-aws/kork-secrets-aws.gradle]: nit: your not using this as far as I can tell.

File: kork-secrets/src/main/java/com/netflix/spinnaker/kork/secrets/EncryptedSecret.java [hunk 2]
```diff
@@ -67,6 +69,7 @@ public static EncryptedSecret parse(String secretConfig) {
   }
 
   protected void update(String secretConfig) {
+    encryptedFile = isEncryptedFile(secretConfig);
```
Comment (fieldju) [inline kork-secrets/src/main/java/com/netflix/spinnaker/kork/secrets/EncryptedSecret.java]: What is this for?

Comment (mayitbeegh) [reply kork-secrets/src/main/java/com/netflix/spinnaker/kork/secrets/EncryptedSecret.java]: a.isFile equivalent, for determine if a config begins with `encryptedFile`

File: kork-secrets-aws/kork-secrets-aws.gradle (hunk 1 above)
Comment (mayitbeegh) [reply kork-secrets-aws/kork-secrets-aws.gradle]: It's only for pulling in junit. I find other modules using spock so it looks more aligned.

**Pull request merged**